pkg/trace/remoteconfighandler/remote_config_handler.go

```diff
@@ -89,13 +91,64 @@ func (h *RemoteConfigHandler) Start() {
 		return
 	}
 
-	h.remoteClient.Start()
-	h.remoteClient.Subscribe(state.ProductAPMSampling, h.onUpdate)
-	h.remoteClient.Subscribe(state.ProductAgentConfig, h.onAgentConfigUpdate)
+	h.client.Start()
+	h.client.Subscribe(state.ProductAPMSampling, h.onUpdate)
+	h.client.Subscribe(state.ProductAgentConfig, h.onAgentConfigUpdate)
+	if h.mrfClient != nil {
+		h.mrfClient.Start()
+		h.mrfClient.Subscribe(state.ProductAgentFailover, h.mrfUpdateCallback)
+	}
+}
+
+// mrfUpdateCallback is the callback function for the AGENT_FAILOVER configs.
+// It fetches all the configs targeting the agent and applies the failover_apm settings.
+// Note: we only care about APM (failover_apm) configuration, but Logs (failover_logs) and Metrics (failover_metrics)
+// may also be present on the MRF configuration. These are handled on the core agent RC.
+func (h *RemoteConfigHandler) mrfUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	// If the updates map is empty, we should unset the failover settings if they were set via RC previously
+	if len(updates) == 0 {
+		log.Debugf("no MRF configuration in remote config update payload")
+		return
+	}
+
+	var enableAPM *bool
```

Comment:
Maybe I'm missing something here, why is this a pointer to a bool? It looks like we accept a pointer in on the config file, is it expected that the value there could dynamically change so we need the pointer?

Reply:
It is a pointer to differentiate `nil` values from `false` ones.